### ISSUE TITLE
Additional query parameters to getOne call

### ISSUE THREAD

**Issue by chattago2002** [mevdschee/ra-data-treeql#10]
Hi.

Is there a way to a "dataProvider.getOne" call with additional query parameters?
I need to get details from User table but I'd like to include only certain columns (e.g. include=name,phone).
There's the "meta" parameter to be set but if I write

```
dataProvider.getOne('users', { id: user_id, meta: { include: 'name' } })
```

what is in "meta" isn't added to the call.

How can I do?

**Comment by nkappler** [comment]
Hi,
this library is currently not designed to take arbitrary params, it is meant as a reference dataprovider implementation for [mevdschee/php-crud-api](https://github.com/mevdschee/php-crud-api).
I guess you are trying to integrate [column selection](https://github.com/mevdschee/php-crud-api#column-selection) into the dataprovider, which I agree would be a valuable feature.

You could either [write your own dataprovider](https://marmelab.com/react-admin/DataProviderWriting.html) based on this one, or extend this implementation. I would be very happy to review a pull-request, however I can't promise to implement this myself in the short term.
I might look into it in one or two weeks, if you can wait until then...

**Comment by nkappler** [comment]
Also, where did you get `meta` from? Could you point to some documentation if possible?

**Comment by chattago2002** [comment]
Hi.
I'm using this library with [mevdschee/php-crud-api](https://github.com/mevdschee/php-crud-api) and [marmelab/react-admin](https://github.com/marmelab/react-admin) (currently I'm using v 4.7).

In the official documentation of React Admin for the [useGetOne hook](https://marmelab.com/react-admin/doc/4.7/useGetOne.html) there is the "meta" argument. As written there

> The meta argument is optional. It can be anything you want to pass to the data provider, e.g. a list of fields to show in the result.

I'm using
```
const dataProvider = useDataProvider();
dataProvider.getOne('users', { id: user_id, meta: { include: 'name' } })
```
because I need to make the call inside a "useEffect".

I can't understand if I'm wrong and where, because the call to 'users' is done but no query parameters are added.

**Comment by nkappler** [comment]
> I can't understand if I'm wrong and where, because the call to 'users' is done but no query parameters are added.

You are not wrong, the functionality was just missing until now.
I've just implemented a change and will publish a new minor version to npm soon 🎉

Thanks for your feedback, please let me know if everything is working as you expected or if something else is broken 😊

**Comment by nkappler** [comment]
Just released v1.2.0 on npm

**Comment by chattago2002** [comment]
> Just released v1.2.0 on npm

Thank you very much, great. I will try it as soon as possible but I'm sure it will be perfect.
